include/xtypes/DynamicData.hpp

```diff
@@ -479,7 +480,7 @@ class ReadableDynamicDataRef
     ReadableDynamicDataRef(
             const DynamicType& type,
             uint8_t* source)
-        : type_(type)
+        : type_(type.kind() == TypeKind::ALIAS_TYPE ? static_cast<const AliasType&>(type).get() : type)
```

Comment:
The most inner types should be resolved even with inner `AliasType`. Else, it will fail with an alias of another alias (or alias of alias of alias..., and go on).

Reply:
I moved the parser test to `parser_test.cpp: TEST(IDLParser, alias_test)` and added a new test under `alias_type.cpp: TEST(AliasType, recursive_dynamic_alias_data)`.